agenta-backend/agenta_backend/services/app_manager.py

```diff
@@ -116,15 +116,17 @@ async def update_variant_image(
 
     valid_image = await deployment_manager.validate_image(image)
     if not valid_image:
-        raise ValueError("Image could not be found in registery.")
+        raise ValueError("Image could not be found in registry.")
     deployment = await db_manager.get_deployment_by_objectid(
         app_variant_db.base.deployment
     )
 
     await deployment_manager.stop_and_delete_service(deployment)
     await db_manager.remove_deployment(deployment)
 
-    await deployment_manager.remove_image(app_variant_db.base.image)
+    if os.environ["FEATURE_FLAG"] in ["ee", "oss"]:
+        await deployment_manager.remove_image(app_variant_db.base.image)
```

Comment:
why don't we remove the image in cloud anymore?

Reply:
We do, but it doesn't occur in the app_manager module.

Let me explain what happens... In cloud, when we serve a variant the first time, it builds the image and then does the following:
- tag the built image with ECR details
- create an ECR repository for the image (if it doesn't exist)
- pushes the image to ECR
- remove the local image from the machine it's on (if it exists)

You see that the last step removes the image after pushing it to ECR. What happens in line 127 is that it removes the ECR repository which causes the variant overwrite not to work. That shouldn't be happening, we always need the ECR repository of the variant because when a user overwrites the variant, it updates the ECR repository with the latest tag.

cc: @mmabrouk

Reply:
Upon further investigation, it has been confirmed that the ECR repository is correctly deleted when a user removes an app in the cloud. Therefore, calling the 'remove_image' deployment_manager function when updating the variant image in the cloud is not needed.